File: modules/live-activity-control/src/LiveActivityControlModule.ts
import { requireNativeModule } from 'expo-modules-core';

export type LiveActivityResult = {
  success: boolean;
  message: string;
  activityId?: string;
  elapsedTime?: number;
};

export interface LiveActivityControlModule {
  startActivity(sideOrName: string, babyNameOrIcon?: string): Promise<LiveActivityResult>;
  endActivity(): Promise<LiveActivityResult>;
  updateActivity(side: string): Promise<LiveActivityResult>;
  pauseActivity(activityId?: string): Promise<LiveActivityResult>;
  resumeActivity(activityId?: string): Promise<LiveActivityResult>;
  completeActivity(activityId?: string): Promise<LiveActivityResult>;
  areActivitiesEnabled(): Promise<boolean>;
}

export default requireNativeModule<LiveActivityControlModule>('LiveActivityControl');

File: modules/live-activity-control/ios/LiveActivityControlModule.swift
import ExpoModulesCore
import ActivityKit
import Foundation

public class LiveActivityControlModule: Module {
    private var nursingModule = NursingLiveActivityModule()
    
    public func definition() -> ModuleDefinition {
        Name("LiveActivityControl")

        // Legacy compatibility functions - delegate to new module
        AsyncFunction("startActivity") { (side: String, babyName: String?) -> [String: Any] in
            let activityName = side.capitalized + " Nursing"
            let activityIcon = "🍼"
            return await self.nursingModule.startActivity(activityName: activityName, activityIcon: activityIcon)
        }

        AsyncFunction("endActivity") { () -> [String: Any] in
            return await self.nursingModule.completeActivity(activityId: nil)
        }

        AsyncFunction("updateActivity") { (side: String) -> [String: Any] in
            // For legacy compatibility, treat as restart with new side
            let activityName = side.capitalized + " Nursing"
            let activityIcon = "🍼"
            return await self.nursingModule.startActivity(activityName: activityName, activityIcon: activityIcon)
        }

        AsyncFunction("areActivitiesEnabled") { () -> Bool in
            return self.nursingModule.areActivitiesEnabled()
        }
        
        // Expose new enhanced functions
        AsyncFunction("pauseActivity") { (activityId: String?) -> [String: Any] in
            return await self.nursingModule.pauseActivity(activityId: activityId)
        }
        
        AsyncFunction("resumeActivity") { (activityId: String?) -> [String: Any] in
            return await self.nursingModule.resumeActivity(activityId: activityId)
        }
        
        AsyncFunction("completeActivity") { (activityId: String?) -> [String: Any] in
            return await self.nursingModule.completeActivity(activityId: activityId)
        }
    }
}

